lib/request.js

```diff
@@ -10,6 +10,7 @@ function Request (params, req, query, headers, log, ip, ips, hostname) {
   this.ip = ip
   this.ips = ips
   this.hostname = hostname
+  this.isNotFound = false
```

Comment:
Please avoid creating one more property in this object. This adds a few bytes of memory for each request.

You can use a getter instead.

Reply:
We can’t use a getter here, as we don’t have access to the context object.
The only way would be to have it in `reply.isNotFound`, but I’m not sure is semantically correct.

Reply:
I'm -1 on adding one more property to Request. We have already too many.

Reply:
So are you suggesting to add it as a getter in `reply`?
If so, I'm not sure that `isNotFound` is the best name to use.

Reply:
Yes. It’s ok to change name.

Reply:
@fastify/core @santonocito any ideas?

Reply:
Right now, if we add it to reply, I don't see any improvements respect `reply.statusCode === 404`

We should only add this line in the fourOhFour.js file to set the code since the beginning

```
const onRequest = this[kHooks].onRequest.concat([(request, reply, done) => {
        reply.code(404)
        done()
      }])
```

Reply:
I think it should be possible to add it as a getter to the request with a little refactoring. But I haven't had time to investigate.

Reply:
@Eomm we can’t use the status code, as it is not reliable. If the user has an hook that changes the status code, the user will never know if they are inside a 404 handling or not.
We must use the internal way we are already using for identify that situation.